src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -268,6 +271,16 @@
                 .onItem().call(this::loadProperties);
     }
 
+    // TODO [BG Phase 2] Delete this method
+    public Uni<List<Endpoint>> getDefaultEndpointsStateless(String tenant) {
+        String query = "SELECT e FROM Endpoint e JOIN e.defaults d WHERE d.id.accountId = :accountId";
+
+        return statelessSession.createQuery(query, Endpoint.class)
+                .setParameter("accountId", tenant)
+                .getResultList()
+                .onItem().call(this::loadProperties);
+    }
+
```

Comment:
This method had to be duplicated, are there recommendations when mixing stateless and stateful sessions?

Reply:
I didn't want to use the `StatelessSession` for any REST API call because the stateful session includes tools that are not available with the stateless one (L1 and L2 caches, persistence context, cascading...). So I duplicated the method to run the same query with the stateless session for messages processing and with the stateful session otherwise.

There's no issue with mixing both kinds of sessions, but the stateful one should be favored whenever possible because of its additional features.

Reply:
I've some worries here:
- We end up duplicating more methods (although it could be fixed with a base method that takes the session?) - In this particular case, this method will go away soon anyway.
- We use a stateless method in a rest call without noticing (or the other way around) - Is there some way to pick the right session depending on the context instead of specifying in each *Resources method call?

We could continue with this PR to test if it fixes our issue and give more though on how to follow up.

Reply:
I understand your concerns, we can certainly think about a better way to pick the right session depending on the context. I have something in mind, I'll create another PR for that.

Regarding the duplication, I don't think that's a problem since there's only one case and it will be deleted as soon as behavior groups go live.

Reply:
The duplication will actually also be removed by the following PR.

Reply:
What I had in mind doesn't actually work because `Session` and `StatelessSession` are completely different objects so there's no room for generics there.

We could isolate `StatelessSession` usages in a dedicated `*Resources` class that would only be used during Kafka messages processing and also add comments explaining when `StatelessSession` should or should not be used.